lib/jara/cli.rb

```diff
@@ -0,0 +1,57 @@
+# encoding: utf-8
+
+require 'jara'
+require 'optparse'
+
+module Jara
+  class Cli
+    def initialize(argv)
+      @command = argv.first
+      @argv = argv.drop(1)
+    end
+
+    def run
+      parse_argv(@argv)
+      options = {}
+      options[:archiver] = @archiver if @archiver
+      options[:build_command] = @build_command if @build_command
+      releaser = Jara::Releaser.new(@environment, @bucket, options)
+      case @command
+      when /help|-h/
+        $stderr.puts(option_parser)
+      when 'release'
+        releaser.release
+      when 'build'
+        releaser.build_artifact
+      else
+        $stderr.puts('Unknown command "%s", expected "build" or "release"' % @command)
+        exit(1)
+      end
+    rescue JaraError => e
```

Comment:
Perhaps I'm mistaken, but I think you want to capture `OptionParser::ParseError` or whatever the type is called as well.